pkg/setup/setup.go

```diff
@@ -77,8 +77,9 @@ type HTTPConfig struct {
 	Register func(*http.ServeMux)
 
 	// optional
-	BasicAuth *BasicAuth
-	Shutdown  func(context.Context) error
+	BasicAuth   *BasicAuth
+	Shutdown    func(context.Context) error
+	Interceptor func(h http.Handler) http.Handler
```

Comment:
I'm not entirely convinced that we need this here. We have a lot of http middlewares and they work fine without having dedicated support in the setup. So, why add this here and not just wrap the http server in the frontend service?

Reply:
Yes, I updated the PR to have this fully set up on the frontend service.